File: final/go/options/options.go
package options

import (
	"fmt"

	"project/jobs"
)

type Option struct {
	Job  func()
	Name string
}

type Options []Option

func GetPatientOptions() Options {
	return Options{
		{
			Name: "Salir",
		},
		{
			Name: "Ver lista de doctores",
			Job:  jobs.ViewDoctors,
		},
		{
			Name: "Ver citas disponibles",
			Job:  jobs.ViewAvailableAppointments,
		},
		{
			Name: "Ver citas disponibles por especialidad",
			Job:  jobs.ViewAvailableAppointmentsBySpecialty,
		},
		{
			Name: "Agendar una cita",
			Job:  jobs.ScheduleAppointment,
		},
		{
			Name: "Cancelar una cita",
			Job:  jobs.CancelAppointment,
		},
	}
}

func GetDoctorOptions() Options {
	return Options{
		{
			Name: "Salir",
		},
		{
			Name: "Ver mis citas",
			Job:  jobs.ViewDoctorAppointments,
		},
		{
			Name: "Ver citas ocupadas",
			Job:  jobs.ViewOcupatesDoctorAppointments,
		},
		{
			Name: "Añadir una cita",
			Job:  jobs.AddAppointment,
		},
		{
			Name: "Reagendar una cita",
		},
		{
			Name: "Eliminar una cita",
		},
	}
}

func (o Options) ViewOptions() {
	for i := range o {
		if i == 0 {
			defer fmt.Printf("%d. %s\n", i, o[i].Name)
		} else {
			fmt.Printf("%d. %s\n", i, o[i].Name)
		}
	}
}

File: final/project/options/Options.java
package project.options;

import project.jobs.ViewDoctors;
import project.jobs.ViewAvailableAppointments;
import project.jobs.ViewAvailableAppointmentsBySpecialty;
import project.jobs.ScheduleAppointment;
import project.jobs.CancelAppointment;
import project.jobs.ViewDoctorAppointments;
import project.jobs.ViewOcupatesDoctorAppointments;
import project.jobs.AddAppointment;
import project.jobs.UpdateAppointment;
import project.jobs.RemoveAppointment;

public class Option {
    public Runnable Job;
    public String Name;
}

public class Options {
    private Option[] options;

    public static Options GetPatientOptions() {
        Options options = new Options();
        options.options = new Option[]{
            new Option(){
                Name = "Salir"
            },
            new Option(){
                Name = "Ver lista de doctores",
                Job = ViewDoctors::run
            },
            new Option(){
                Name = "Ver citas disponibles",
                Job = ViewAvailableAppointments::run
            },
            new Option(){
                Name = "Ver citas disponibles por especialidad",
                Job = ViewAvailableAppointmentsBySpecialty::run
            },
            new Option(){
                Name = "Agendar una cita",
                Job = ScheduleAppointment::run
            },
            new Option(){
                Name = "Cancelar una cita",
                Job = CancelAppointment::run
            }
        };
        return options;
    }

    public static Options GetDoctorOptions() {
        Options options = new Options();
        options.options = new Option[]{
            new Option(){
                Name = "Salir"
            },
            new Option(){
                Name = "Ver mis citas",
                Job = ViewDoctorAppointments::run
            },
            new Option(){
                Name = "Ver citas ocupadas",
                Job = ViewOcupatesDoctorAppointments::run
            },
            new Option(){
                Name = "Añadir una cita",
                Job = AddAppointment::run
            },
            new Option(){
                Name = "Reagendar una cita",
                Job = UpdateAppointment::run
            },
            new Option(){
                Name = "Eliminar una cita",
                Job = RemoveAppointment::run
            }
        };
        return options;
    }

    public void ViewOptions() {
        for (int i = 0; i < options.length; i++) {
            if (i == 0) {
                System.out.printf("%d. %s\n", i, options[i].Name);
            } else {
                System.out.printf("%d. %s\n", i, options[i].Name);
            }
        }
    }
}
